src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
You have to be very careful about using loops within loops (filter is one loop and find is another). There is something called Big O Notation and this is a way to describe the performance of an algorithm.
You can read more there: https://hackernoon.com/the-big-o-notation-in-javascript

In your case its O(N^2). And "N" there describe our people array length so 39.

Better idea is to first find all of mothers and then check is some of them has children